File: in1000/host2020/uke2/busstur.py
# 02.20: Busstur

plasser = 30
passasjerer = 0 #maks 30, kan ikke overstige antall plasser
nye_passasjerer = 0
stasjon_nummer = 1

nye_passasjerer = int(input("Stasjon {}! Hvor mange gaar paa bussen?: ".format(stasjon_nummer)))
if (nye_passasjerer <= plasser):
    passasjerer += nye_passasjerer
    plasser -= passasjerer
    stasjon_nummer += 1
    print(nye_passasjerer, "gaar ombord.")
else:
    passasjerer += nye_passasjerer
    plasser -= passasjerer
    print("Bussen er full!", passasjerer-30, "maa gaa til fots.")
    quit()

nye_passasjerer = int(input("Stasjon {}! Hvor mange gaar paa bussen?: ".format(stasjon_nummer)))
if (nye_passasjerer <= plasser):
    passasjerer += nye_passasjerer
    plasser -= passasjerer
    stasjon_nummer += 1
    print(nye_passasjerer, "gaar ombord.")
else:
    passasjerer += nye_passasjerer
    plasser -= passasjerer
    print("Bussen er full!", passasjerer-30, "maa gaa til fots.")
    quit()

nye_passasjerer = int(input("Stasjon {}! Hvor mange gaar paa bussen?: ".format(stasjon_nummer)))
if (nye_passasjerer <= plasser):
    passasjerer += nye_passasjerer
    plasser -= passasjerer
    stasjon_nummer += 1
    print(nye_passasjerer, "gaar ombord.")
else:
    passasjerer += nye_passasjerer
    plasser -= passasjerer
    print("Bussen er full!", passasjerer-30, "maa gaa til fots.")

print("Bussen kjorer til endestasjonen med", passasjerer, "passasjerer.")

File: inf1000/uke1/Busstur.java
/**
1.14: Busstur
*/

import java.util.Scanner;

class Busstur {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int plasser = 30;
        int passasjerer = 0;
        int i = 0;

        for (i = i; i < 3; i++) {
            if (plasser > 0) {
                System.out.print("Stasjon " + (i+1) + "! Hvor mange gaar paa bussen?: ");
                plasser = plasser - (Integer.parseInt(in.next()));

                if (plasser < 0) {
                    System.out.println("Bussen er full! " + (plasser - passasjerer)*(-1) + " maa gaa.");
                }
            }
        }
    }
}
